src/App.tsx

```diff
@@ -15,48 +16,140 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SortByAlphabet = 'alphabet',
+  SortByLength = 'length',
+  DefaultValue = '',
+}
+
+interface SortingOptions {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+function getSortedGoodsList(
+  goods: string[],
+  { sortType, isReversed }: SortingOptions,
+) {
+  const sortedGoodList = [...goods];
+
+  if (sortType) {
+    sortedGoodList.sort((a, b) => {
+      switch (sortType) {
+        case SortType.SortByLength: {
+          return a.length - b.length;
+        }
+
+        case SortType.SortByAlphabet: {
+          return a.localeCompare(b);
+        }
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedGoodList.reverse();
+  }
+
+  return sortedGoodList;
+}
+
 export const App: React.FC = () => {
+  const [sortingOptions, setSortingOptions] = useState<SortingOptions>({
+    sortType: SortType.DefaultValue,
+    isReversed: false,
+  });
+
+  const visibleGoodsList = getSortedGoodsList(
+    goodsFromServer,
+    sortingOptions,
+  );
+
+  const handlerSetSortType = (sortType: SortType) => () => {
```

Comment:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364
```suggestion
  const handleSortTypeChange = (sortType: SortType) => () => {
```